lib/cocoapods/installer.rb

```diff
@@ -317,7 +317,7 @@ def create_and_save_projects(pod_targets_to_generate, aggregate_targets_to_gener
         projects_by_pod_targets = pod_project_generation_result.projects_by_pod_targets
 
         predictabilize_uuids(generated_projects) if installation_options.deterministic_uuids?
-        stabilize_target_uuids(generated_projects)
+        stabilize_target_uuids(generated_projects) if installation_options.incremental_installation?
```

Comment:
We're seeing the issue with incremental installation flag on - this will not help for #681